Assets/_BForBoss/_Weapons/Scripts/Weapons/MeleeWeaponBehaviour.cs

```diff
@@ -0,0 +1,101 @@
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace Perigon.Weapons
+{
+    
+    public interface IMeleeWeapon
+    {
+        float CurrentCooldown { get; }
+        float MaxCooldown { get; }
+        bool CanMelee { get; }
+    }
+    
+    public class MeleeWeaponBehaviour : MonoBehaviour, IMeleeWeapon
+    {
+        [SerializeField] private MeleeScriptableObject _meleeScriptable;
+        [SerializeField] private Transform _playerTransform;
+
+        [SerializeField] private bool _canAttackMany = true;
+
+        private MeleeWeapon _weapon;
+        private InputAction _meleeActionInputAction;
+
+        public float CurrentCooldown => _weapon?.CurrentCooldown ?? 0f;
+        public float MaxCooldown => _meleeScriptable != null ? _meleeScriptable.AttackCoolDown : 1f;
+        public bool CanMelee => _weapon?.CanMelee ?? false;
+
+        public void Initialize(InputAction meleeAttackAction, IMeleeProperties properties = null)
```

Comment:
btw, just a suggestions if you want to implement, or if you think it's worth. Can also inject the `Transform` into the Initialize
`(() => Transform)`

if you don't want `MeleeWeaponBehaviour` to be serialized

Reply:
Ah good idea, I'll make that change

Reply:
Made this change - I still had to leave it serialized for the gizmo to work, but I think its still a worthwhile improvement just cause its more portable this way